homeassistant/components/huawei_lte/sensor.py

```diff
@@ -46,453 +46,577 @@
+                key="dl_mcs",
+                name="Downlink MCS",
+                entity_category=EntityCategory.DIAGNOSTIC,
+            ),
+            "dlbandwidth": HuaweiSensorEntityDescription(
+                key="dlbandwidth",
+                name="Downlink bandwidth",
+                icon_fn=lambda x: (
+                    "mdi:speedometer-slow",
+                    "mdi:speedometer-medium",
+                    "mdi:speedometer",
+                )[bisect((8, 15), x if x is not None else -1000)],
+                entity_category=EntityCategory.DIAGNOSTIC,
+            ),
+            "earfcn": HuaweiSensorEntityDescription(
+                key="earfcn",
+                name="EARFCN",
+                entity_category=EntityCategory.DIAGNOSTIC,
+            ),
+            "ecio": HuaweiSensorEntityDescription(
+                key="ecio",
+                name="EC/IO",
+                device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+                # https://wiki.teltonika.lt/view/EC/IO
+                icon_fn=lambda x: (
+                    "mdi:signal-cellular-outline",
+                    "mdi:signal-cellular-1",
+                    "mdi:signal-cellular-2",
+                    "mdi:signal-cellular-3",
+                )[bisect((-20, -10, -6), x if x is not None else -1000)],
+                state_class=SensorStateClass.MEASUREMENT,
+                entity_category=EntityCategory.DIAGNOSTIC,
+            ),
+            "enodeb_id": HuaweiSensorEntityDescription(
+                key="enodeb_id",
+                name="eNodeB ID",
+                entity_category=EntityCategory.DIAGNOSTIC,
+            ),
+            "lac": HuaweiSensorEntityDescription(
+                key="lac",
+                name="LAC",
+                icon="mdi:map-marker",
+                entity_category=EntityCategory.DIAGNOSTIC,
+            ),
+            "ltedlfreq": HuaweiSensorEntityDescription(
+                key="ltedlfreq",
+                name="Downlink frequency",
+                formatter=lambda x: (
+                    round(int(x) / 10) if x is not None else None,
+                    UnitOfFrequency.MEGAHERTZ,
+                ),
+                device_class=SensorDeviceClass.FREQUENCY,
+                entity_category=EntityCategory.DIAGNOSTIC,
+            ),
+            "lteulfreq": HuaweiSensorEntityDescription(
+                key="lteulfreq",
+                name="Uplink frequency",
+                formatter=lambda x: (
+                    round(int(x) / 10) if x is not None else None,
+                    UnitOfFrequency.MEGAHERTZ,
+                ),
+                device_class=SensorDeviceClass.FREQUENCY,
+                entity_category=EntityCategory.DIAGNOSTIC,
+            ),
+            "mode": HuaweiSensorEntityDescription(
+                key="mode",
+                name="Mode",
+                formatter=lambda x: (
+                    {"0": "2G", "2": "3G", "7": "4G"}.get(x, "Unknown"),
```

Comment:
We use `None` to represent unknown state.

Reply:
Included in https://github.com/home-assistant/core/pull/84019